### PR TITLE
[PT-186] Signup form endpoint

### PR DESCRIPTION
**Pull request by 0xbuilder0**
# [PT-186](https://0xbase.atlassian.net/browse/PT-186) - Signup form endpoint

### Description

[PT-186]: https://0xbase.atlassian.net/browse/PT-186?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by 0xbuilder0** [comment]
TODO: swagger docs

**Comment by 0xbuilder0** [comment]
> TODO: swagger docs

DONE

File: internal/routes/portfolios.go [hunk 1]
```diff
@@ -5,31 +5,36 @@ import (
...
+	"github.com/0xbase-Corp/portfolio_svc/internal/middlewares"
 )
 
 var PortfolioRoutes = func(router *gin.Engine, db *gorm.DB) {
 	router.GET("/healthy", controllers.HealthCheck)
 
 	v1 := router.Group("/api/v1")
 
-	v1.GET("/portfolio/solana", func(c *gin.Context) { controllers.SolanaController(c, db) })
+	v1.GET("/portfolio/solana", middlewares.VerifyToken(), func(c *gin.Context) { controllers.SolanaController(c, db) })
```
Comment (fuddin95) [inline internal/routes/portfolios.go:16]: Please group middleware like so https://gin-gonic.com/docs/examples/using-middleware/

File: shared/utils/generate-token.go [hunk 2]
```diff
@@ -10,22 +10,20 @@ import (
...
 		Issuer    string `json:"iss"`
 		Audience  string `json:"aud"`
 		IssuedAt  int64  `json:"iat"`
 	}
 )
 
-func GenerateAccessToken(userID int, email, publicKey string) (string, error) {
-	return generateToken(userID, email, publicKey, 438000*time.Hour)
+func GenerateAccessToken(userID string) (string, error) {
+	return generateToken(userID, 12*time.Minute)
```
Comment (fuddin95) [inline shared/utils/generate-token.go:22]: Please create a const something `ACCESS_TOKEN_EXPIRY`  & `REFRESH_TOKEN_EXPIRY`

**Review by fuddin95 — APPROVED** [review]

**Review by fuddin95 — CHANGES_REQUESTED** [review]
Ping once done

File: shared/utils/generate-token.go (hunk 2 above)
Comment (0xbuilder0) [reply shared/utils/generate-token.go]: done

File: internal/routes/portfolios.go (hunk 1 above)
Comment (0xbuilder0) [reply internal/routes/portfolios.go]: done

File: internal/routes/portfolios.go [hunk 3]
```diff
@@ -5,31 +5,35 @@ import (
...
-	v1.POST("/generate-hash", func(c *gin.Context) { controllers.AuthGenerateHash(c, db) })
-
-	v1.POST("/verify-hash", func(c *gin.Context) { controllers.AuthVerifyHashKey(c, db) })
-
-	v1.POST("/bundles", func(c *gin.Context) { controllers.CreateBundleController(c, db) })
-
-	v1.GET("/bundles/:id", func(c *gin.Context) { controllers.GetBundlesByIdController(c, db) })
-
+	// Routes that don't require authentication
+	auth := router.Group("/api/v1/auth")
```
Comment (fuddin95) [inline internal/routes/portfolios.go:15]: we just keep the path `"/api/v1"` no need for "auth"

**Review by fuddin95 — APPROVED** [review]

File: internal/routes/portfolios.go (hunk 3 above)
Comment (0xbuilder0) [reply internal/routes/portfolios.go:15]: just group the routes like signup and signin is group by `/api/v1/auth/XXXX` same for portfolios like `/api/v1/portfolios/XXXX`  and for bundles like `/api/v1/bundles/XXXX`

**Pull request merged**